File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/loanchargedialog/LoanChargeDialogMvpView.java
package com.mifos.mifosxdroid.dialogfragments.loanchargedialog;

import com.mifos.mifosxdroid.base.MvpView;
import com.mifos.objects.client.ChargeCreationResponse;

import okhttp3.ResponseBody;

/**
 * Created by Rajan Maurya on 9/6/16.
 */
public interface LoanChargeDialogMvpView extends MvpView {

    void showAllChargesV3(ResponseBody response);

    void showLoanChargesCreatedSuccessfully(ChargeCreationResponse chargeCreationResponse);

    void showChargeCreatedFailure(String errorMessage);

    void showError(String errorMessage);
}

File: mifosng-android/src/main/java/com/mifos/mifosxdroid/dialogfragments/loanchargedialog/LoanChargeDialogMvpView.kt
package com.mifos.mifosxdroid.dialogfragments.loanchargedialog

import com.mifos.mifosxdroid.base.MvpView
import com.mifos.objects.client.ChargeCreationResponse
import okhttp3.ResponseBody

/**
 * Created by Rajan Maurya on 9/6/16.
 */
interface LoanChargeDialogMvpView : MvpView {
    fun showAllChargesV3(response: ResponseBody)
    fun showLoanChargesCreatedSuccessfully(chargeCreationResponse: ChargeCreationResponse)
    fun showChargeCreatedFailure(errorMessage: String)
    fun showError(errorMessage: String)
}